packages/core-commands/src/admin-navigation-commands.js

```diff
@@ -25,9 +46,24 @@ export function useAdminNavigationCommands() {
 	useCommand( {
 		name: 'core/manage-reusable-blocks',
 		label: __( 'Manage all of my patterns' ),
-		callback: () => {
-			document.location.href = 'edit.php?post_type=wp_block';
+		callback: ( { close } ) => {
+			// __unstableIsBlockBasedTheme is not defined in the site editor so we need to check the context.
+			// If not the site editor and not a block based theme then redirect to the old wp-admin patterns page.
+			if ( ! isSiteEditor && ! isBlockTheme ) {
+				document.location.href = 'edit.php?post_type=wp_block';
+			} else {
+				const args = {
+					path: '/patterns',
+				};
+				const targetUrl = addQueryArgs( 'site-editor.php', args );
+				if ( isSiteEditor ) {
+					history.push( args );
+				} else {
+					document.location = targetUrl;
+				}
+				close();
+			}
```

Comment:
Should we also be checking the user has the required capabilities to access the site editor? See https://github.com/WordPress/gutenberg/pull/52403

Reply:
Done.